test/e2e/cloud/nodes.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"time"
+
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("Nodes", func() {
+	f := framework.NewDefaultFramework("cloudprovider")
+	var c clientset.Interface
+
+	BeforeEach(func() {
+		// Only supported in AWS/GCE because those are the only cloud providers
+		// where E2E test are currently running.
+		framework.SkipUnlessProviderIs("aws", "gce", "gke")
+		c = f.ClientSet
+	})
+
+	It("should be deleted on API server if it doesn't exist in the cloud provider", func() {
+		By("deleting a node on the cloud provider")
+
+		nodeDeleteCandidates := framework.GetReadySchedulableNodesOrDie(c)
+		nodeToDelete := nodeDeleteCandidates.Items[0]
+
+		origNodes := framework.GetReadyNodesIncludingTaintedOrDie(c)
+		framework.Logf("Original number of ready nodes: %d", len(origNodes.Items))
+
+		err := framework.DeleteNodeOnCloudProvider(&nodeToDelete)
+		if err != nil {
+			framework.Failf("failed to delete node %q, err: %q", nodeToDelete.Name, err)
+		}
+
+		newNodes, err := framework.CheckNodesReady(c, len(origNodes.Items)-1, 5*time.Minute)
+		Expect(err).To(BeNil())
+		Expect(len(newNodes)).To(Equal(len(origNodes.Items) - 1))
+		Expect(checkNodeExistsInList(newNodes, nodeToDelete)).To(BeFalse())
```

Comment:
How about checking the nonexistence of the node on Node API like
```
_, err := client.CoreV1().Nodes().Get(nodeToDelete.Name, metav1.GetOptions{})
if err == nil {
        framework.Failf("The node %q still exists", nodeToDelete.Name)
} else if !errors.IsNotFound(err) {
        framework.Failf("failed to get node %q, err: %q", nodeToDelete.Name, err)
}
```
instead if line 55-70?

Reply:
I think this can replace line 58, but I would still want to check total # of nodes incase another node was accidentally deleted. What do you think?

Reply:
That seems a nice idea for me :-)